src/Microsoft.AspNet.Mvc.TagHelpers/Internal/ModeMatchResult.cs

```diff
@@ -63,17 +64,16 @@ public class ModeMatchResult<TMode>
                     match => match.PresentAttributes.Any(
                         attribute => PartiallyMatchedAttributes.Contains(
                             attribute, StringComparer.OrdinalIgnoreCase)));
-
-                logger.LogWarning(new PartialModeMatchLogValues<TMode>(uniqueId, viewPath, partialOnlyMatches));
+                //TODO: What the heck is this thing?
```

Comment:
A few of our `TagHelper`s have various modes that they can run on. A great example is `ScriptTagHelper`. Some of the things it can for is:

**Include multiple `src` files:** `<script asp-src-include="*.js"></script>`
**Fallback to `src`:** 
```HTML
<script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.1.4.min.js"
		asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
		asp-fallback-test="window.jQuery">
```

Or a combination.

Therefore, we created a system to determine a `TagHelper`s mode that could be re-used across several of our `TagHelper`s.